server-core/engine/src/main/java/io/infectnet/server/engine/core/world/strategy/AStarPathFinderStrategy.java

```diff
@@ -0,0 +1,298 @@
+package io.infectnet.server.engine.core.world.strategy;
+
+import io.infectnet.server.engine.core.world.Position;
+import io.infectnet.server.engine.core.world.Tile;
+import io.infectnet.server.engine.core.world.World;
+
+import java.util.ArrayList;
+import java.util.LinkedList;
+import java.util.List;
+
+import static io.infectnet.server.engine.core.world.TileType.CAVE;
+
+/**
+ * Class for Path finding in the world between two Positions, using the A* algorithm.
+ */
+public class AStarPathFinderStrategy implements PathFinderStrategy{
+
+    /**
+     * A heuristic to use with value calculating.
+     */
+    private final Heuristic heuristic;
+
+    /**
+     * The world, in which we want to find the path.
+     */
+    private World world;
+
+    /**
+     * The destination Position.
+     */
+    private Position targetPos;
+
+    /**
+     * The list of Nodes that we do not yet consider fully searched.
+     */
+    private List<Node> openNodes;
+
+    /**
+     * Creates a startegy with the given heuristic to use in the A* algorithm.
+     * @param heuristic the specific heuristic function to use
+     */
+    public AStarPathFinderStrategy(Heuristic heuristic){
+        this.heuristic = heuristic;
+    }
+
+    @Override
+    public LinkedList<Tile> findPath(World world, Position start, Position target) {
+
+        /* Initialization */
+
+        this.world = world;
+        targetPos = target;
+        openNodes = new ArrayList<>();
```

Comment:
`openNodes` and `closedNodes` should be `HashSet`s instead of `List`s. This way the `contains` method which is used very frequently will have constant complexity that results in a potential performance gain.